rl_chain/rl_chain_base.py

```diff
@@ -199,55 +201,53 @@ def _learn(self, vw_ex):
         self.workspace.learn_one(multi_ex)
 
 
+def embed_string_type(item: Union[str, _Embed], model: Any, default_namespace: Optional[str] = None) -> Dict[str, str]:
+    """Helper function to embed a string or an _Embed object."""
+    if isinstance(item, _Embed):
+        encoded = model.encode(item.impl)
+        join_char = " "
+    else:
+        encoded = item
+        join_char = ""
+
+    if default_namespace is None:
+        raise ValueError("The default namespace must be provided when embedding a string or _Embed object.")
+    
+    return {default_namespace: join_char.join(map(str, encoded))}
+
+def embed_dict_type(item: Dict, model: Any) -> Dict[str, str]:
+    """Helper function to embed a dictionary item."""
+    inner_dict = {}
+    for ns, embed_str in item.items():
+        inner_dict.update(embed_string_type(embed_str, model, ns))
+    return inner_dict
+
 def embed(
-    to_embed: Union[str, Dict, List[str], List[Dict]],
+    to_embed: Union[Union(str, _Embed(str)), Dict, List[Union(str, _Embed(str))], List[Dict]],
```

Comment:
[nit] looks splitting into _embed(Union[str, _Embed(str), Dict]) and embed(...) can simplify the code (we just call _embed in list case without caring about type) - but maybe not very much